docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+    - `refactor test runner`
+    - `document API endpoints`
+- Use lowercase letters
+- Keep it concise but clear (aim for under 50 characters)
+- Bad examples:
+    - `changes` (too vague)
+    - `Added new feature` (past tense)
+    - `WIP` (uninformative)
+    - `Fix stuff` (too vague)
+
+For bigger changes, you can add more details after the first line.
+
+> [!TIP]
+> Remember: A good commit message lets other developers know what changed
+without having to look at the code!
+
+### 4. Pull Requests
+Pull requests are how you get your changes merged into the codebase. 
+You create a pull request once you feel your changes are ready.
+
+#### Creating a Pull Request
+- Make sure you have pushed all changes to GitHub first
+- The title of your pull request should be simple, concise, and descriptive
+  - It should have a prefix just like an issue
+  - It can be basically the name of the branch
+    - Issue: `Frontend: refactor all server calls to one place`
+    - Branch: `server-communicator`
+    - Pull Request: `Frontend: ServerCommunicator`
+- Fill out the Pull Request template with enough detail for others to understand what 
+  changed without having to read all the new code.
+- Make sure all GitHub actions pass:
+  - Your pull request will be blocked from merging if all tests don't pass
+- Move the connected issue into the `PR Submittted` column on the 
+  [TA Project Board](https://github.com/orgs/softwareconstruction240/projects/1/views/9)
+- Request a review from other Autograder developers
+  - If you're changing an existing system, request a review from the developer that 
+    wrote it initially, or is most familiar with that system
+  - You can request multiple reviewers, but you're only required to get approval from one
+
+#### Scope of a Pull Request
+- Pull Requests should represent a complete, reviewable unit of work
+- In most cases, a pull request from one issue
+  - Multiple issues in one PR are fine and encouraged when they depend on each other, or form a cohesive change
+- While there's no strict size limit, consider breaking up very large changes if:
+  - They touch multiple unrelated areas
+  - They're becoming difficult to review
+  - They could logically be shipped separately
+
+> [!TIP]
+> Remember: A good PR represents a complete feature or fix, but shouldn't try to solve everything at once!
+
+#### Reviewing a Pull Request
+A crucial part of this process is the code review. The repo has been set up so that _**no changes may be merged 
+into main without the review of at least one other developer.**_ If someone has requested a review from you, please
+take some time to help out and ensure the Autograder has quality code.
+
+**As a Reviewer**
+- Look for:
+  - Code quality and style guide compliance
+  - Test coverage
+  - Potential bugs or edge cases
+  - Documentation completeness
+- Checkout the branch on your local machine and do some manual testing
+  - Manually check systems that changed
```

Comment:
Merge this line with the previous one.

Reply:
Conversely, split the previous line into two separate lines. Emphasize that changed systems should be explicitly verified in addition to generally testing that "nothing broke."

Reply:
I split the line